hail_scripts/v01/utils/elasticsearch_client.py

```diff
@@ -103,8 +103,33 @@ def export_vds_to_elasticsearch(
                 "sample_id: s",  # add sample_id to each genotype struct to keep track of
             ])
 
-            genotypes_root = "gs" if not discard_missing_genotypes else "gs.filter(g => g.isCalled())"
-            vds = vds.annotate_variants_expr("va.genotypes = %(genotypes_root)s.map(g => { %(genotype_struct_expr)s }).collect()" % locals())
+            vds = vds.annotate_variants_expr("va.genotypes = gs.map(g => { %(genotype_struct_expr)s }).collect()" % locals())
+
+            sample_fields_annotation_template = "va.samples_%(field_name)s = va.genotypes.filter(gen => %(field_filter)s).map(gen => gen.sample_id).toSet"
+            for i in range(1, 3):
+                vds = vds.annotate_variants_expr(sample_fields_annotation_template % dict(
+                    field_name='num_alt_{}'.format(i), field_filter='gen.num_alt == {}'.format(i)
+                ))
+            vds = vds.annotate_variants_expr(sample_fields_annotation_template % dict(
+                field_name='no_call', field_filter='gen.num_alt == -1'
+            ))
+
+            for i in range(5, 95, 5):
+                vds = vds.annotate_variants_expr(sample_fields_annotation_template % dict(
+                    field_name='gq_gte_{}'.format(i), field_filter='gen.gq >= {0} && gen.gq < {0} + 5'.format(i)
```

Comment:
Would it worth adding the upper bound to the field names to clarify what they contain?

Reply:
maybe, but I already loaded a dataset using this schema that took 4 days and wrote the seqr code to parse it, and i'm not really willing to do that again